qiskit/circuit/quantumcircuit.py

```diff
@@ -140,6 +140,9 @@ class QuantumCircuit:
     def __init__(self, *regs, name=None, global_phase=0):
         if any([not isinstance(reg, (QuantumRegister, ClassicalRegister)) for reg in regs]):
             try:
+                if any([not isinstance(reg, (int, np.int, np.int32, np.int64)) for reg in regs]):
```

Comment:
I'm not sure about that list of types. What about something like this?
```suggestion
                if any([int(reg) != reg for reg in regs]):
```
and raise an exception instead of a warning.

Reply:
Thanks for the feedback.  That would be better. 

Though, what would you think of using `math.isclose(int(reg), reg)` to handle numbers that have picked up small floating point errors?